### PR TITLE
samples: matter: Refactored samples configuration structure

### PR DESCRIPTION
**Pull request by kkasperczyk-no**
Refactored samples configuration structure to utilize
Matter upstream Kconfig defaults for application and child images

Signed-off-by: Kamil Kasperczyk <kamil.kasperczyk@nordicsemi.no>

### CODE REVIEW

File: doc/nrf/releases/release-notes-changelog.rst [hunk 1]
```diff
@@ -289,6 +289,8 @@ Matter samples
   * :ref:`matter_lock_sample` sample:
 
     * Added support for the Door Lock cluster, which replaced the previous temporary solution based on the On/Off cluster.
+  
+  * Simplified all samples :file:`prj.conf` files by using default configuration from Matter upstream :file:`Kconfig.defaults` file.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:293]: Move this to line 274.
```suggestion
  * Simplified the :file:`prj.conf` file of each sample by using the default configuration from the :file:`Kconfig.defaults` file in Matter upstream.
```

File: doc/nrf/releases/release-notes-changelog.rst [hunk 2]
```diff
@@ -701,6 +703,7 @@ The Matter fork in the |NCS| (``sdk-connectedhomeip``) contains all commits from
 The following list summarizes the most important changes inherited from the upstream Matter:
 
 * Added the Binding cluster and Groupcast communication to the Light Switch sample.
+* Added for nRF Connect SDK platform :file:`Kconfig.defaults`, :file:`Kconfig.mcuboot.defaults` and :file:`Kconfig.multiprotocol_rpmsg.defaults` files containing default configuration common for all nRF Connect SDK samples.
```
Comment (greg-fer) [inline doc/nrf/releases/release-notes-changelog.rst:706]:
```suggestion
* Updated the nRF Connect platform by adding :file:`Kconfig.defaults`, :file:`Kconfig.mcuboot.defaults` and :file:`Kconfig.multiprotocol_rpmsg.defaults` files that contain the default configuration for all nRF Connect SDK samples.
```
The platform in Matter is called `nRF Connect` last time I checked, not `nRF Connect SDK`, but maybe you meant something else here than my suggestion? Let me know.

**Review by greg-fer — CHANGES_REQUESTED** [review]
Are these changes bringing in any potential changes to the way Matter is configured in the NCS?

File: samples/matter/light_bulb/CMakeLists.txt [hunk 3]
```diff
@@ -6,7 +6,11 @@
 
 cmake_minimum_required(VERSION 3.20.0)
 
-set(APPLICATION_CONFIG_DIR "configuration/\${BOARD}")
+set(MATTER_MODULE_ROOT $ENV{ZEPHYR_BASE}/../modules/lib/matter)
```
Comment (Damian-Nordic) [inline samples/matter/light_bulb/CMakeLists.txt:9]:
```suggestion
get_filename_component(MATTER_MODULE_ROOT $ENV{ZEPHYR_BASE}/../modules/lib/matter REALPATH)
```
to use shorter paths.

**Review by Damian-Nordic — APPROVED** [review]

**Comment by kkasperczyk-no** [comment]
> Are these changes bringing in any potential changes to the way Matter is configured in the NCS?

@greg-fer I'm not sure what changes exactly you mean, but this change overall changes the values that user need to set in the prj.conf. Currently we have plenty of duplicates and setting dozen of configs that are in fact the same for every Matter sample. That's why we created Kconfig.defaults that sets all configs common for all Matter samples and in prj.confs only the configuration specific for given sample will be necessary.

**Comment by greg-fer** [comment]
> Currently we have plenty of duplicates and setting dozen of configs that are in fact the same for every Matter sample. That's why we created Kconfig.defaults that sets all configs common for all Matter samples and in prj.confs only the configuration specific for given sample will be necessary.

OK, @kkasperczyk-no , maybe we could include this information on the configuration page https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_matter_configuring_protocol.html
See my suggestions on Teams.

File: doc/nrf/ug_matter_configuring_protocol.rst [hunk 4]
```diff
@@ -23,7 +23,14 @@ Because Matter is an application layer protocol on top of the other IPv6-based t
 It uses modules such as Bluetooth® LE, the IPv6 stack (currently, only :ref:`Thread <ug_thread_configuring>` is supported), :ref:`nRF Security <nrfxlib:nrf_security_config>`, or :ref:`MCUboot <mcuboot:mcuboot_ncs>`.
 Make sure to review the configuration options of these modules when configuring Matter.
 
-For an example configuration, see the :ref:`Matter Template sample's <matter_template_sample>` :file:`prj.conf` files in the :file:`configuration` directory for each supported board target.
+The Kconfig options for Matter applications in the nRF Connect SDK are stored in the following files:
+
+* :file:`prj.conf` files, which are specific to the application.
+* :file:`Kconfig.defaults` file, which is available in the :file:`module/lib/matter/config/nrfconnect/chip-module` directory and is used to populate :file:`prj.conf` with Kconfig option settings common to all samples.
+
+The Matter samples use the same structure for other software images, such as mcuboot or multiprotocol_rpmsg, which have default options defined in the corresponding :file:`Kconfig.mcuboot.defaults` and :file:`Kconfig.multiprotocol_rpmsg.defaults` files.
```
Comment (greg-fer) [inline doc/nrf/ug_matter_configuring_protocol.rst:31]:
```suggestion
The Matter samples use the same structure for other software images, such as MCUboot or Multiprotocol RPMsg, which have default options defined in the corresponding :file:`Kconfig.mcuboot.defaults` and :file:`Kconfig.multiprotocol_rpmsg.defaults` files.
```

**Review by greg-fer — APPROVED** [review]
Approving, but please fix the capitalization from comment.

**Pull request merged**